presto-docs/src/main/sphinx/gcp/storage.rst

```diff
@@ -0,0 +1,216 @@
+        <value>777</value>
+        <description>
+          Optional. Override the default '700' simulated POSIX file permissions.
+        </description>
+    </property>
+
+More configuration properties and their default values are shown in `gcs-core-default.xml <https://github.com/GoogleCloudPlatform/bigdata-interop/blob/v1.9.10/gcs/conf/gcs-core-default.xml>`_.
+
+Hive Connector
+^^^^^^^^^^^^^^
+
+Another requirement is that you have enabled and configured a Hive connector in Presto.
+The connector uses Hive metastore for data discovery and is not limited to data residing on HDFS.
+
+**Configuring Hive Connector**
+
+* URL to Hive metastore:
+
+    * New Hive metastore on GCP:
+
+        If your Presto nodes are provisioned by GCP, your Hive metastore should also be on GCP to minimize latency and costs.
+        The simplest way to create a new Hive metastore on GCP is to create a small Cloud DataProc cluster (1 master, 0 workers), accessible from your Presto cluster.
+        Follow the steps for existing Hive metastore after finishing this step.
+
+    * Existing Hive metastore:
+
+        To use an existing Hive metastore with a Presto cluster, you need to set the ``hive.metastore.uri`` property in your Hive catalog properties file to ``thrift://${METASTORE_ADDRESS}:${METASTORE_THRIFT_PORT}``.
+        If the metastore uses authentication, please refer to `Hive Security Configuration`_.
+
+        .. _Hive Security Configuration: https://docs.starburstdata.com/latest/connector/hive-security.html
+
+* GCS access:
+
+    * Simplified configuration:
+
+        Here are example values for all GCS configuration properties which can be set in the Hive catalog properties file:
+
+        .. code-block:: properties
+
+            # JSON key file used to access Google Cloud Storage
+            hive.gcs.json-key-file-path=/path/to/gcs_keyfile.json
+
+            # Use client-provided OAuth token to access Google Cloud Storage
+            hive.gcs.use-access-token=false
+
+    * Full configuration:
+
+        If you need more configuration options, finish the `Full GCS configuration`_ section and then add the full path to your ``core-site.xml`` file to the ``hive.config.resources`` property in the Hive catalog properties file.
+        This will allow Presto Hive connector to pick up GCS configuration during startup.
+
+Accessing GCS Data From Presto for the First Time
+-------------------------------------------------
+
+Accessing Data Already Mapped in the Hive metastore
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you migrate to Presto from Hive, chances are that your GCS data is already mapped to SQL tables in the metastore. 
+In that case, you should be able to query it.
+
+Accessing Data Not Yet Mapped in the Hive metastore
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To access GCS data that is not yet mapped in the Hive metastore you need to provide the schema of the data, the file format, and the data location. 
```

Comment:
We already have an existing example for creating an external table in S3. What do you think about expanding that with more text like you have here? Or maybe pull that out into a separate section?